sfs/mono/drivingfunction.py

```diff
@@ -739,37 +1322,36 @@ def esa_edge_dipole_2d_line(omega, x0, xs, alpha=3/2*np.pi, Nc=None, c=None):
 
     Returns
     -------
-    (N,) numpy.ndarray
+    d : (N,) numpy.ndarray
         Complex weights of secondary sources.
+    selection : (N,) numpy.ndarray
+        Boolean array containing ``True`` or ``False`` depending on
+        whether the corresponding secondary source is "active" or not.
+    secondary_source_function : callable
+        A function that can be used to create the sound field of a
+        single secondary source.  See `sfs.mono.synthesize()`.
+
+    Notes
+    -----
+    One leg of the secondary sources has to be located on the x-axis (y0=0),
+    the edge at the origin.
+
+    Derived from :cite:`Spors2016`
 
     """
     x0 = np.asarray(x0)
-    k = util.wavenumber(omega, c)
-    phi_s = np.arctan2(xs[1], xs[0])
-    if phi_s < 0:
-        phi_s = phi_s + 2*np.pi
-    r_s = np.linalg.norm(xs)
-    L = x0.shape[0]
-
-    r = np.linalg.norm(x0, axis=1)
-    phi = np.arctan2(x0[:, 1], x0[:, 0])
-    phi = np.where(phi < 0, phi+2*np.pi, phi)
-
-    if Nc is None:
-        Nc = np.ceil(2 * k * np.max(r) * alpha/np.pi)
-
-    epsilon = np.ones(Nc)  # weights for series expansion
-    epsilon[0] = 2
+    xs = np.asarray(xs)
+    xref = np.asarray(xref)
 
-    d = np.zeros(L, dtype=complex)
-    idx = (r <= r_s)
-    for m in np.arange(Nc):
-        nu = m*np.pi/alpha
-        f = 1/epsilon[m] * np.cos(nu*phi_s) * np.cos(nu*phi)
-        d[idx] = d[idx] + f[idx] * jn(nu, k*r[idx]) * hankel2(nu, k*r_s)
-        d[~idx] = d[~idx] + f[~idx] * jn(nu, k*r_s) * hankel2(nu, k*r[~idx])
+    if np.isscalar(xref):
+        a = np.linalg.norm(xref)/np.linalg.norm(xref-xs)
+    else:
+        a = np.linalg.norm(xref-x0, axis=1)/np.linalg.norm(xs-xref)
```

Comment:
It is kinda hard to see in this diff but you seem to have changed `xref-xs` to `xs-xref`.
I guess it doesn't matter if you calculate the norm of the resulting vector, but I was still wondering if this was intentional?

Reply:
was intentionally to check a broadcast issue, forgot to recover this, sorry. Fixed.